From 60ea1ce6e8f5e8b258e15a2f7eaf10b9563e8614 Mon Sep 17 00:00:00 2001
From: Harish Mahendrakar <hmahendrakar@google.com>
Date: Fri, 5 Jan 2024 21:57:48 +0000
Subject: [PATCH] Disable building software OMX codecs

Bug: 316432618
Test: Build and verify that libstagefright_soft_* are not buit.

Change-Id: I59868b984fc3807b1e5cbb9c2180e05f1790222d
---
 media/libstagefright/omx/Android.bp |  3 +++
 services/mediacodec/Android.bp      | 25 -------------------------
 2 files changed, 3 insertions(+), 25 deletions(-)

diff --git a/media/libstagefright/omx/Android.bp b/media/libstagefright/omx/Android.bp
index 54c5697c146..79ab009ffb8 100644
--- a/media/libstagefright/omx/Android.bp
+++ b/media/libstagefright/omx/Android.bp
@@ -184,6 +184,9 @@ cc_library_shared {
 
 cc_defaults {
     name: "libstagefright_softomx-defaults",
+    // TODO (b/316432618) Software OMX codecs are no longer used, disable building them till
+    // this code is removed completely.
+    enabled: false,
     vendor_available: true,
 
     cflags: [
diff --git a/services/mediacodec/Android.bp b/services/mediacodec/Android.bp
index a2f17c22b9b..506b3bca679 100644
--- a/services/mediacodec/Android.bp
+++ b/services/mediacodec/Android.bp
@@ -95,31 +95,6 @@ cc_binary {
         "android.hidl.memory@1.0",
     ],
 
-    runtime_libs: [
-        "libstagefright_soft_aacdec",
-        "libstagefright_soft_aacenc",
-        "libstagefright_soft_amrdec",
-        "libstagefright_soft_amrnbenc",
-        "libstagefright_soft_amrwbenc",
-        "libstagefright_soft_avcdec",
-        "libstagefright_soft_avcenc",
-        "libstagefright_soft_flacdec",
-        "libstagefright_soft_flacenc",
-        "libstagefright_soft_g711dec",
-        "libstagefright_soft_gsmdec",
-        "libstagefright_soft_hevcdec",
-        "libstagefright_soft_mp3dec",
-        "libstagefright_soft_mpeg2dec",
-        "libstagefright_soft_mpeg4dec",
-        "libstagefright_soft_mpeg4enc",
-        "libstagefright_soft_vorbisdec",
-        "libstagefright_soft_vpxdec",
-        "libstagefright_soft_vpxenc",
-        "libstagefright_softomx_plugin",
-    ],
-
     // OMX interfaces force this to stay in 32-bit mode;
     compile_multilib: "32",
